### ISSUE TITLE
Spark SQL support

### ISSUE THREAD

**Issue by lcpinheiro** [darwin-eu-dev/CdmOnboarding#43]
Could you please add Spark SQL support? The function `cdmOnboarding()` throws an error related to a few specificities of the Spark SQL flavour:

DBMS:
spark

Error:
java.sql.SQLException: [Simba][SparkJDBCDriver](500051) ERROR processing query/statement. Error Code: 0, SQL state: org.apache.hive.service.cli.HiveSQLException: Error running query: org.apache.spark.sql.catalyst.parser.ParseException: 
missing ')' at 'Name'(line 4, pos 28)

== SQL ==
SELECT *
FROM (
 select ROW_NUMBER() OVER(ORDER BY COUNT(device_exposure_id) DESC) AS ROW_NUM,
 Cr.concept_name as Concept Name,
----------------------------^^^
 floor((COUNT(device_exposure_id)+99)/100)*100 as  floor((COUNT(distinct person_id)+99)/100)*100 as  from hive_metastore.imrd_uk_202205.device_exposure C
JOIN hive_metastore.imrd_uk_202205.CONCEPT CR
ON C.device_concept_id = CR.CONCEPT_ID
where c. device_concept_id != 0
group by CR.concept_name
having COUNT(device_exposure_id)>5
) z
WHERE z.ROW_NUM <= 25
ORDER BY z.ROW_NUM

**Comment by lcpinheiro** [comment]
To be clear, two fails. The alias with a space "concept name" fails, using concept_name or `concept name` works. count_big fails, even if the sqlRender package has the conversion "spark,COUNT_BIG(@a),COUNT(@a)" - it should have translated correctly but didn't

**Comment by MaximMoinat** [comment]
Thanks for reporting, we are working on a v2 with some rewrites of above mentioned queries and will take your comments into account.

Once v2 is ready for testing, would you be able to execute the new code on your Spark database? Thanks!

**Comment by MaximMoinat** [comment]
Note: this is a bug in SqlRender, where I opened an issue as well https://github.com/OHDSI/SqlRender/issues/308

**Comment by MaximMoinat** [comment]
@lcpinheiro Fix has been implemented in SqlRender, and this should be supported in upcoming release of SqlRender

**Comment by MaximMoinat** [comment]
To confirm: SqlRender 1.11.0 has been released with this fix.

@lcpinheiro Could you update SqlRender with `remotes::install_github('OHDSI', 'SqlRender')` and try to run CdmOnboarding again? I am very interested whether the issue is indeed solved.

**Comment by lcpinheiro** [comment]
Hey @MaximMoinat, the SQL translation seems to have been addressed. I bumped into another issue. I got an rds file in my output folder, but with an error. Is this ignorable? Should I open another issue?

2022-12-12 13:54:15	[Main thread]	INFO	CdmOnboarding	.execute	Complete CdmOnboarding took 15.91 minutes
2022-12-12 13:54:15	[Main thread]	INFO	CdmOnboarding	.execute	The CDM Onboarding results have been exported to: cdm_onboarding_res
2022-12-12 13:54:17	[Main thread]	ERROR	CdmOnboarding	3	Could not generate results document: Error in order(df_dm$`#RECORDS`, decreasing = TRUE): argument 1 is not a vector 
2022-12-12 13:54:17	[Main thread]	INFO	CdmOnboarding	3	Results from the checks have been saved as an RDS object to the output folder.

**Comment by MaximMoinat** [comment]
Was there an error while executing the drug mapping query?

Btw, v2 is almost ready to be released. If you want, you could already try that by using ref=release-v2 in remotes::install_github

**Comment by PRijnbeek** [comment]
V2 is released

**Comment by PRijnbeek** [comment]
Seems like a drive error again. Will try to replicate this with v2 on my Databricks instance.

**Comment by lcpinheiro** [comment]
Hi @PRijnbeek, we've heard from Databricks support. They confirmed the suspicions that that the driver does some SQL translation, and the suggested hack is adding the argument: `UseNativeQuery=1` to the connection string. 

This seems to have sorted the SQL statement error for the `CdmOnboarding` package and I now have generated an onboarding report of one of our databases. 

There is one error thrown that is odd, below, regarding the zip file (below). The other error `cannot open connection` error I assume is because I had the file open at the time this was generated.


`Warning: Failed to bundle CDM Onboarding results, no zip bundle has been created: Error in .jcall("RJavaTools", "Ljava/lang/Object;", "invokeMethod", cl, : java.lang.RuntimeException: java.io.FileNotFoundException: C:\Projects\OMOP test\cdm_onboarding_res\cdm_onboarding_res\Results_Onboarding_IMRD UK.zip (The system cannot find the path specified)  Warning messages: 1: In value[[3L]](cond) :   Error 'cannot open the connection' when writing log to file 'cdm_onboarding_res/log_cdmOnboarding.txt. Removing file appender from logger. 2: In normalizePath(path.expand(path), winslash, mustWork) :   path[1]="cdm_onboarding_res/CdmOnboarding-IMRD UK.docx": The system cannot find the path specified 3: In normalizePath(path.expand(path), winslash, mustWork) :   path[1]="cdm_onboarding_res/log_cdmOnboarding.txt": The system cannot find the path specified 4: In normalizePath(path.expand(path), winslash, mustWork) :   path[1]="cdm_onboarding_res/mapping_levels_drugsErr.txt": The system cannot find the path specified 5: In normalizePath(path.expand(path), winslash, mustWork) :   path[1]="cdm_onboarding_res/onboarding_results_IMRD UK.rds": The system cannot find the path specified Error in rJava::.jcall(metaData, "Ljava/sql/ResultSet;", "getCatalogs") :    java.sql.SQLException: [Simba][SparkJDBCDriver](500003) Error getting catalog information: Metadata Initialization Error.`

**Comment by PRijnbeek** [comment]
I see twice \cdm_onboarding_res in that path.

**Comment by lcpinheiro** [comment]
Indeed, well spotted, not sure why:

```
outputFolder <- "cdm_onboarding_res"

results <- 
  CdmOnboarding::cdmOnboarding(
  connectionDetails = connectionDetails,
  cdmDatabaseSchema = cdmDatabaseSchema,
  resultsDatabaseSchema = resultsDatabaseSchema,
  vocabDatabaseSchema = vocabDatabaseSchema,
  databaseId = databaseId,
  databaseName = databaseName,
  databaseDescription = databaseDescription,
  runVocabularyChecks = TRUE,
  runDataTablesChecks = TRUE,
  runPerformanceChecks = TRUE,
  runWebAPIChecks = FALSE,
  smallCellCount = smallCellCount,
  sqlOnly = FALSE,
  outputFolder = outputFolder,
  verboseMode = verboseMode
) 
```

**Comment by MaximMoinat** [comment]
Zipping also fails sometimes when testing, but this is not reproducible. More investigation is needed.

However, zipping is more a convenience step and not blocking. All output files are generated and can be zipped manually if needed. I will adapt the error message to reflect that.
